lib/index.ts

```diff
@@ -0,0 +1,1114 @@
+        return true
+      } else if (checkAllAliases(key, flags.nargs)) {
+        error = Error(__('Invalid configuration: %s, opts.count excludes opts.narg.', key))
+        return true
+      }
+      return false
+    })
+  }
+
+  return {
+    argv: Object.assign(argvReturn, argv),
+    error: error,
+    aliases: Object.assign({}, flags.aliases),
+    newAliases: Object.assign({}, newAliases),
+    defaulted: Object.assign({}, defaulted),
+    configuration: configuration
+  }
+}
+
+// if any aliases reference each other, we should
+// merge them together.
+function combineAliases (aliases: Dictionary<string | string[]>): Dictionary<string[]> {
+  const aliasArrays: Array<string[]> = []
+  const combined: Dictionary<string[]> = Object.create(null)
+  let change = true
+
+  // turn alias lookup hash {key: ['alias1', 'alias2']} into
+  // a simple array ['key', 'alias1', 'alias2']
+  Object.keys(aliases).forEach(function (key) {
+    aliasArrays.push(
+      ([] as string[]).concat(aliases[key], key)
+    )
+  })
+
+  // combine arrays until zero changes are
+  // made in an iteration.
+  while (change) {
+    change = false
+    for (let i = 0; i < aliasArrays.length; i++) {
+      for (let ii = i + 1; ii < aliasArrays.length; ii++) {
+        const intersect = aliasArrays[i].filter(function (v) {
+          return aliasArrays[ii].indexOf(v) !== -1
+        })
+
+        if (intersect.length) {
+          aliasArrays[i] = aliasArrays[i].concat(aliasArrays[ii])
+          aliasArrays.splice(ii, 1)
+          change = true
+          break
+        }
+      }
+    }
+  }
+
+  // map arrays back to the hash-lookup (de-dupe while
+  // we're at it).
+  aliasArrays.forEach(function (aliasArray) {
+    aliasArray = aliasArray.filter(function (v, i, self) {
+      return self.indexOf(v) === i
+    })
+    const lastAlias = aliasArray.pop()
+    if (lastAlias !== undefined && typeof lastAlias === 'string') {
+      combined[lastAlias] = aliasArray
+    }
+  })
+
+  return combined
+}
+
+// this function should only be called when a count is given as an arg
+// it is NOT called to set a default value
+// thus we can start the count at 1 instead of 0
+function increment (orig?: number | undefined): number {
+  return orig !== undefined ? orig + 1 : 1
+}
+
+// TODO(bcoe): in the next major version of yargs, switch to
+// Object.create(null) for dot notation:
+function sanitizeKey (key: string): string {
+  if (key === '__proto__') return '___proto___'
+  return key
+}
+
+const yargsParser: Parser = function Parser (args: ArgsInput, opts?: Partial<Options>): Arguments {
+  const result = parse(args.slice(), opts)
+  return result.argv
+}
+
+// parse arguments and return detailed
+// meta information, aliases, etc.
+yargsParser.detailed = function (args: ArgsInput, opts?: Partial<Options>): DetailedArguments {
+  return parse(args.slice(), opts)
+}
+
+export { yargsParser }
```

Comment:
let's make `yargsParser` the `default export (this goes along with dropping the `index.js`).